### PR TITLE
Update to k8s 1.20

### PR DESCRIPTION
**Pull request by surajssd**

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # vi: set ft=ruby :
 
 Vagrant.configure("2") do |config|
-  config.vm.box = "ubuntu/bionic64"
+  config.vm.box = "ubuntu/hirsute64"
```
Comment (invidian) [inline Vagrantfile]: I think we should use LTS version where possible.

File: scripts/setup-etcd [hunk 2]
```diff
@@ -29,5 +29,6 @@ cp \
 systemctl daemon-reload
 systemctl enable etcd
 systemctl start etcd
+systemctl status --no-pager etcd
```
Comment (invidian) [inline scripts/setup-etcd:32]: Why is this needed? Please add some explanation in commit message.

File: certificates/kube-scheduler-csr.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (invidian) [inline certificates/kube-scheduler-csr.json:1]: Can you please add some explanation why this was missing before?

File: scripts/setup-controller-services [hunk 4]
```diff
@@ -44,7 +44,7 @@ systemctl enable kube-apiserver kube-controller-manager kube-scheduler
 systemctl start kube-apiserver kube-controller-manager kube-scheduler
 EOF
 
-  until vagrant ssh "controller-${i}" -- curl --ipv4 --fail --silent --max-time 3 "http://127.0.0.1:8080/api" &>/dev/null; do
+  until vagrant ssh "controller-${i}" -- curl --ipv4 --fail --silent --max-time 3 -k "https://localhost:6443/healthz" &>/dev/null; do
```
Comment (invidian) [inline scripts/setup-controller-services:49]: I guess the advantage is to run checks over HTTPS and use more up to date endpoint? Can we mention that in the commit message please?

File: scripts/setup-controller-services [hunk 5]
```diff
@@ -40,8 +40,10 @@ cp \
   /etc/systemd/system/kube-controller-manager.service
 
 systemctl daemon-reload
-systemctl enable kube-apiserver kube-controller-manager kube-scheduler
-systemctl start kube-apiserver kube-controller-manager kube-scheduler
+services="kube-apiserver kube-controller-manager kube-scheduler"
```
Comment (invidian) [inline scripts/setup-controller-services:43]: Ditto commit message explanation. I think something like: "... so user can immediately see that services are running (or functional/not crashing, something like this)"

File: scripts/generate-kubelet-config-file [hunk 6]
```diff
@@ -17,6 +17,7 @@ authentication:
     clientCAFile: "/var/lib/kubernetes/ca.pem"
 authorization:
   mode: Webhook
+cgroupDriver: systemd
```
Comment (invidian) [inline scripts/generate-kubelet-config-file:20]: Is this default with latest Ubuntu? It might not be the case for LTS version.

File: scripts/install-tools [hunk 7]
```diff
@@ -20,13 +20,13 @@ trap 'popd; rm -rf "${tmpdir}"' EXIT
 source "${dir}/versions.bash"
 
 curl -sSL \
-  -O "https://pkg.cfssl.org/${cfssl_version}/cfssl_${binary_platform}-${binary_format}" \
```
Comment (invidian) [inline scripts/install-tools:23]: Please add some comment in commit message on why is pkg.cfssl.org no longer a good source.

File: Vagrantfile [hunk 8]
```diff
@@ -17,7 +17,6 @@ Vagrant.configure("2") do |config|
       c.vm.provision :shell, :path => "scripts/vagrant-setup-haproxy.bash"
 
       c.vm.provider "virtualbox" do |vb|
-        vb.memory = "256"
```
Comment (invidian) [inline Vagrantfile:20]: Why?

File: Vagrantfile [hunk 9]
```diff
@@ -28,7 +28,7 @@ Vagrant.configure("2") do |config|
         c.vm.provision :shell, :path => "scripts/vagrant-setup-hosts-file.bash"
 
         c.vm.provider "virtualbox" do |vb|
-          vb.memory = "640"
+          vb.memory = "750"
```
Comment (invidian) [inline Vagrantfile:31]: 750MB still seem a little for controller node. Did you experience some issues with 640MB? Can you please mention that in commit message if so?

File: manifests/coredns.yaml [hunk 10]
```diff
@@ -75,44 +79,29 @@ metadata:
...
-              matchExpressions:
-              - key: k8s-app
-                operator: In
-                values: ["kube-dns"]
-            topologyKey: kubernetes.io/hostname
       containers:
       - name: coredns
-        image: coredns/coredns:1.6.7
-        imagePullPolicy: IfNotPresent
+        image: quay.io/kinvolk/coredns:1.8.0
```
Comment (invidian) [inline manifests/coredns.yaml]: Why not use `coredns/coredns:1.8.0`?

File: README.md [hunk 11]
```diff
@@ -168,7 +168,7 @@ Deploy the DNS add-on and verify it's working:
 ```
 kubectl apply -f ./manifests/coredns.yaml
 [...]
-kubectl get pods -l k8s-app=kube-dns -n kube-system
+kubectl get pods -l k8s-app=coredns -n kube-system
```
Comment (invidian) [inline README.md:196]: This should probably go with config update commit.

**Review by invidian — COMMENTED** [review]
Overall looks good to me. I would be curious about the reasoning behind some of the proposed changes, which is not immediately obvious to me.

File: scripts/generate-kubelet-config-file (hunk 6 above)
Comment (surajssd) [reply scripts/generate-kubelet-config-file:20]: This is set by the runtime provider. That's how we detect what to set for Kubelet by reading the docker's output. What's the benefit of one over other? systemd vs cgroupfs?

File: scripts/install-tools (hunk 7 above)
Comment (surajssd) [reply scripts/install-tools:23]: I don't think it is a bad source, but github gives more fine grained control over the version.

File: scripts/generate-kubelet-config-file (hunk 6 above)
Comment (invidian) [reply scripts/generate-kubelet-config-file:20]: `cgroupfs` is a legacy one (v1) IIRC and `systemd` is v2 (unified). So ideally you would use `systemd`.

File: scripts/install-tools (hunk 7 above)
Comment (invidian) [reply scripts/install-tools:23]: IIRC on GitHub you also have more recent version. Please mention those in commit message :)

File: Vagrantfile [hunk 12]
```diff
@@ -2,7 +2,7 @@
 # vi: set ft=ruby :
```
Comment (invidian) [inline Vagrantfile:2]: Nit:
```diff
-Vagrantfile: Update VMs to latest ubuntu
+Vagrantfile: Update VMs to latest Ubuntu
```

File: scripts/install-tools (hunk 7 above)
Comment (surajssd) [reply scripts/install-tools:23]: Yep, that's done.

File: scripts/generate-kubelet-config-file (hunk 6 above)
Comment (surajssd) [reply scripts/generate-kubelet-config-file:20]: It worked fine on the LTS version as well.

**Review by invidian — APPROVED** [review]
OK
